app/components/ui/PeoplePicker/PeoplePickerLogic.js

```diff
@@ -0,0 +1,123 @@
+/* eslint-disable react/no-unused-prop-types */
+import React from 'react'
+import PropTypes from 'prop-types'
+import { escapeRegExp } from 'lodash'
+
+class PeoplePickerLogic extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      selection: this.props.initialSelection,
+      searchValue: '',
+    }
+  }
+
+  toggleSelection(person) {
+    if (this.isSelected(person)) {
+      this.setState({
+        selection: this.state.selection.filter(p => p.id !== person.id),
+      })
+    } else {
+      this.setState({ selection: this.state.selection.concat(person) })
+    }
+  }
+
+  handleSubmit() {
+    if (this.isValid()) {
+      this.props.onSubmit(this.state.selection)
+    }
+  }
+
+  isSelected(person) {
+    return this.state.selection.some(p => p.id === person.id)
+  }
+
+  isValid() {
+    const selectionLength = this.state.selection.length
+    return (
+      selectionLength >= this.props.minSelection &&
+      selectionLength <= this.props.maxSelection
+    )
+  }
+
+  searchSubmit = searchValue => {
+    this.setState({ searchValue })
+  }
+
+  filterPeople = (people, searchValue, field) => {
+    if (!searchValue) return people
+
+    const inputValue = searchValue.trim().toLowerCase()
+    if (!inputValue) return people
+
+    return people.filter(person =>
+      person[field].toLowerCase().includes(inputValue),
+    )
+  }
+
+  getMatchIndex = (inputValue, option) => {
+    const re = new RegExp(escapeRegExp(inputValue))
+    const match = re.exec(option.toLowerCase())
+    if (match) return match.index
+    return -1
+  }
+
+  render() {
+    const { people, ...otherProps } = this.props
+    let extendedPeople = [...people].sort((a, b) =>
+      a.name.localeCompare(b.name),
+    )
+
+    extendedPeople = extendedPeople.map(person => ({
+      ...person,
+      searchValue: `${person.name} ${person.focuses.join(
+        ' ',
+      )} ${person.expertises.join(' ')} ${person.aff ? person.aff : ''}`,
+    }))
+    const searchOptions = extendedPeople.map(person => ({
+      value: person.name,
+    }))
+    return this.props.children({
+      ...otherProps,
+      people: this.filterPeople(
+        extendedPeople,
+        this.state.searchValue,
+        'searchValue',
+      ),
+      searchSubmit: this.searchSubmit,
+      searchOptions,
+      filterFunction: this.filterPeople,
+      getMatchIndex: this.getMatchIndex,
+      isSelected: person => this.isSelected(person),
+      isValid: this.isValid(),
+      selection: this.state.selection,
+      onSubmit: () => this.handleSubmit(),
+      toggleSelection: person => this.toggleSelection(person),
+    })
+  }
+}
+
+const peopleArrayPropType = PropTypes.arrayOf(
+  PropTypes.shape({
+    id: PropTypes.oneOfType([PropTypes.number, PropTypes.string]).isRequired,
+    name: PropTypes.string.isRequired,
+    aff: PropTypes.string.isRequired,
```

Comment:
Does this propType make the following ternary unnecessary?
https://github.com/elifesciences/elife-xpub/blob/c6817a581f39f197a159205254db63048d174c8b/app/components/ui/PeoplePicker/PeoplePickerLogic.js#L75

Reply:
Also, is there any chance the app may correctly encounter a Person entry that doesn't have a defined value for `aff`, `expertise` or `focuses`?I know these aren't required fields in the Person elife-api schema.